### PR TITLE
Add API entries to allowed urls

### LINKED ISSUE #1014: Orphan check fails for API docs
`npm run check:orphan-pages -- ---historical-apis --current-apis --dev-apis` fails because it needs more entries in the ignore list.

### PR DESCRIPTION
**Pull request by kaelynj**
Closes #1014 

In the interest of readability, the list of allowed urls includes pages that don't exist.  There is a small risk of a false positive if there are any new pages that get added to the api docs of the same name which should actually be checked against their toc instead of ignored.

### CODE REVIEW

File: scripts/commands/checkOrphanPages.ts [hunk 1]
```diff
@@ -74,6 +75,7 @@ async function main() {
       ),
     );
   }
+  console.log("\n Allowed Orphans are: \n", ALLOWED_ORPHAN_URLS);
```
Comment (Eric-Arellano) [inline scripts/commands/checkOrphanPages.ts:78]: Leftover? This is noisy 

```suggestion
```

File: scripts/commands/checkOrphanPages.ts [hunk 2]
```diff
@@ -121,6 +123,7 @@ async function determineTocFiles(args: Arguments): Promise<string[]> {
     globs.push(
       "docs/api/{qiskit,qiskit-ibm-provider,qiskit-ibm-runtime}/[0-9]*/_toc.json",
     );
+    ``;
```
Comment (Eric-Arellano) [inline scripts/commands/checkOrphanPages.ts:126]: Tip before opening up PRs to review your PR proactively to check for issues like this

```suggestion
```

File: scripts/commands/checkOrphanPages.ts [hunk 3]
```diff
@@ -139,4 +142,67 @@ function collectTocFileContents(children: TocEntry[]): string[] {
...
+    "0.41/",
+    "0.42/",
+    "0.43/",
+    "0.44/",
+    "0.45/",
+    "0.46/",
+  ];
+
+  return flattenDeep([
+    versions.map((vers) => [
```
Comment (Eric-Arellano) [inline scripts/commands/checkOrphanPages.ts:184]: Instead, use `.flatMap()` here. Then you can remove `flattenDeep` :)

File: scripts/commands/checkOrphanPages.ts [hunk 4]
```diff
@@ -139,4 +142,67 @@ function collectTocFileContents(children: TocEntry[]): string[] {
...
+      `/api/qiskit/${vers}qiskit.optimization.INFINITY`,
+      `/api/qiskit/${vers}qiskit.quantum_info.two_qubit_cnot_decompose`,
+      `/api/qiskit/${vers}qiskit.utils.algorithm_globals`,
+      `/api/qiskit/${vers}parallel`,
+      `/api/qiskit/${vers}transpiler_builtin_plugins`,
+    ]),
+    `/api/qiskit/0.19/index`,
+    `/api/qiskit/dev/qiskit.primitives.BaseEstimator`,
+    `/api/qiskit/dev/qiskit.primitives.BaseSampler`,
+    `/api/qiskit/dev/qiskit.primitives.BaseSampler`,
```
Comment (Eric-Arellano) [inline scripts/commands/checkOrphanPages.ts:204]: This doesn't fail on the current API docs? Also BaseSampler is repeated

**Review by Eric-Arellano — COMMENTED** [review]
Thanks!

File: scripts/commands/checkOrphanPages.ts (hunk 4 above)
Comment (kaelynj) [reply scripts/commands/checkOrphanPages.ts:204]: Nope, but it does fail on the dev version

File: scripts/commands/checkOrphanPages.ts (hunk 1 above)
Comment (kaelynj) [reply scripts/commands/checkOrphanPages.ts]: Fixed!

File: scripts/commands/checkOrphanPages.ts (hunk 2 above)
Comment (kaelynj) [reply scripts/commands/checkOrphanPages.ts]: Duly noted, I appreciate your patience in reviewing these :)

File: scripts/commands/checkOrphanPages.ts (hunk 3 above)
Comment (kaelynj) [reply scripts/commands/checkOrphanPages.ts]: Good suggestion, thanks!

File: scripts/commands/checkOrphanPages.ts [hunk 5]
```diff
@@ -180,28 +178,25 @@ function apiDocsIgnores(): string[] {
...
+    `/api/qiskit/${vers}ibmq_visualization`,
+    `/api/qiskit/${vers}qiskit.aqua.aqua_globals`,
+    `/api/qiskit/${vers}qiskit.optimization.INFINITY`,
+    `/api/qiskit/${vers}qiskit.quantum_info.two_qubit_cnot_decompose`,
+    `/api/qiskit/${vers}qiskit.utils.algorithm_globals`,
+    `/api/qiskit/${vers}parallel`,
+    `/api/qiskit/${vers}transpiler_builtin_plugins`,
     `/api/qiskit/0.19/index`,
     `/api/qiskit/dev/qiskit.primitives.BaseEstimator`,
     `/api/qiskit/dev/qiskit.primitives.BaseSampler`,
```
[the remark this answers is not in the document]
Comment (kaelynj) [reply scripts/commands/checkOrphanPages.ts]: Fixed now!

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**